ldclient/sse_client.py

```diff
@@ -46,7 +48,12 @@ def _connect(self):
 
         # Use session if set.  Otherwise fall back to requests module.
         requester = self.session or requests
-        self.resp = requester.get(self.url, stream=True, **self.requests_kwargs)
+        self.resp = requester.get(
+            self.url,
+            stream=True,
+            timeout=(self._connect_timeout, self._read_timeout),
```

Comment:
Setting the read_timeout here means that the request will throw a timeout error if no data is received for 5 minutes.. thus implementing heartbeat detection.